### ISSUE TITLE
Installation on OS X/macOS results in Blank Page

### ISSUE THREAD

**Issue by jango-fx** [snipe/snipe-it#3321]
I tried installing Snipe-It on 3 different OS X machines now, after several Error 403s and 500s, I'm finally at a point, where I don't get any error messages anymore, but no working Snipe-It installation either. I'm afraid I completely ran out of ideas by now...

#### Expected Behavior

Seeing the preflight/setup page

-----

#### Actual Behavior

I see a blank white page. No error messages.

-----

#### Please confirm you have done the following before posting your bug report:

- [x] I have enabled debug mode
- [x] I have read [checked the Common Issues page](https://snipe-it.readme.io/docs/common-issues)
- [x] I've searched for similar issues

-----
#### Please provide answers to these questions before posting your bug report:

- current GitHub master branch (as of 20th february 2017)
- macOS Sierra 10.12.4
- Apache/2.4.23
- PHP 5.6.29
- I tried a manual installation like so:

**Installed MariaDB**
`brew install mariadb`

**Set Up Database**
```
mysql.server start
mysql -u root
CREATE DATABASE snipeit;
CREATE USER 'snipe_user'@'localhost' IDENTIFIED BY 'password‘;
GRANT ALL ON snipeit.* TO 'snipe_user'@'localhost‘ IDENTIFIED BY 'password‘;
```

**Downloaded Snipe-It**
```
cd /Library/WebServer/Documents/
sudo git clone https://github.com/snipe/snipe-it
cd snipe_it

cp .env.example .env
chown -R $USER storage public/uploads
chmod -R 755 storage
chmod -R 755 public/uploads
```

**Installed Composer**
```
curl -sS https://getcomposer.org/installer | php
php composer.phar install --no-dev --prefer-source
php artisan key:generate
```

**Configured Snipe-It**

`nano .env`

```
# --------------------------------------------
# REQUIRED: BASIC APP SETTINGS
# --------------------------------------------
APP_ENV=production
APP_DEBUG=true
APP_KEY=base64:IrpF6uO8m8BZljn0UKPnwmW4H9x/BvYo7danJ9lX6X0=
APP_URL=http://localhost
APP_TIMEZONE='UTC'
APP_LOCALE=en


# --------------------------------------------
# REQUIRED: DATABASE SETTINGS
# --------------------------------------------
DB_CONNECTION=mysql
DB_HOST=localhost
DB_DATABASE=snipeit
DB_USERNAME=snipe_user
DB_PASSWORD=password
DB_PREFIX=null
DB_DUMP_PATH='/usr/bin'
```

**Set Up Apache WebServer**
`sudo nano /etc/apache2/httpd.conf`

CHANGED:
`#LoadModule php5_module libexec/apache2/libphp5.so`
→ `LoadModule php5_module libexec/apache2/libphp5.so`

CHANGED:
`#LoadModule rewrite_module libexec/apache2/mod_rewrite.so`
→ `LoadModule rewrite_module libexec/apache2/mod_rewrite.so`

CHANGED:
`#Include /private/etc/apache2/extra/httpd-vhosts.conf`
→ `Include /private/etc/apache2/extra/httpd-vhosts.conf`

`sudo nano /private/etc/apache2/extra/httpd-vhosts.conf`

ADDED:
```
<VirtualHost *:80>
    <Directory "/Library/WebServer/Documents/snipe-it/public/">
    	Allow From All
    	AllowOverride All
    	Options +Indexes
    </Directory>
    DocumentRoot /Library/WebServer/Documents/snipe-it/public/
    ServerName http://localhost
    ServerAlias www.localhost
</VirtualHost>
```

`sudo apachectl restart`

If I now try to load either:
  - http://localhost/
  - http://localhost/public
  - http://localhost/public/index.php

I just get blank white page. Without any error messages. Nothing in the console either.

The Apache error log says:

> [Mon Feb 20 18:42:36.478884 2017] [mpm_prefork:notice] [pid 35948] AH00169: caught SIGTERM, shutting down
> AH00558: httpd: Could not reliably determine the server's fully qualified domain name, using MacBookPro.local. Set the 'ServerName' directive globally to suppress this message
> [Mon Feb 20 18:42:36.801994 2017] [mpm_prefork:notice] [pid 35987] AH00163: Apache/2.4.23 (Unix) PHP/5.6.29 configured -- resuming normal operations
> [Mon Feb 20 18:42:36.802068 2017] [core:notice] [pid 35987] AH00094: Command line: '/usr/sbin/httpd -D FOREGROUND'

`/Library/WebServer/Documents/snipe-it/storage/logs` is empty.

What did I miss??

**Comment by jango-fx** [comment]
anything?

**Comment by snipe** [comment]
Need to know what's in your laravel log. Blank white page is usually a permissions issue. https://snipe-it.readme.io/docs/common-issues#section-browser-redirects-to-login-but-page-is-blank-with-a-500-error

**Comment by jango-fx** [comment]
ok, I changed permissions: `chmod -R 777 storage`
i get a proper error message and log files ^^

> PDOException in Connector.php line 55: SQLSTATE[HY000] [2002] No such file or directory

[laravel.txt](https://github.com/snipe/snipe-it/files/838686/laravel.txt)

**Comment by snipe** [comment]
Please see #3381 and #2676, and this SO article: http://stackoverflow.com/questions/19475762/setting-up-laravel-on-a-mac-php-artisan-migrate-error-no-such-file-or-directory

It sounds like it's looking for a unix socket, and you're not giving it one.

**Comment by jango-fx** [comment]
uhm, not sure what did the trick in the end, but after fiddling around with the mysql setup and a few  restarts I managed to progress to an "Redirect Error" for the setup.

That was resolved by changing `APP_URL` from `http://127.0.0.1/snipe-it` to `http://127.0.0.1`.

Now it seems to run.
Phew! That was probably not as complicated as i made it look...

THANKS VERY MUCH for help and patience! in retrospective, definitely one of my less bright moments...
